client/trino-client/src/main/java/io/trino/client/FixJsonDataUtils.java

```diff
@@ -56,28 +56,29 @@ final class FixJsonDataUtils
 {
     private FixJsonDataUtils() {}
 
-    public static Iterable<List<Object>> fixData(List<Column> columns, List<List<Object>> data)
+    public static Iterable<List<Object>> fixData(List<Column> columns, List<List<Object>> rows)
     {
-        if (data == null) {
+        if (rows == null) {
             return null;
         }
         ColumnTypeHandler[] typeHandlers = createTypeHandlers(columns);
-        ImmutableList.Builder<List<Object>> rows = ImmutableList.builderWithExpectedSize(data.size());
+        // Jackson deserializes lists to ArrayList's so instead of copying lists, update them in place
```

Comment:
this is fragile

Reply:
but well tested - if this changes in some future jackson release - we will know ;)